src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -288,6 +288,13 @@ private void SolutionEvents_AfterClosing()
                     {
                         shouldStartNewBGJobRunner = false;
                     }
+                    else if (_lockService.Value.IsLockHeld)
```

Comment:
this blocked all auto restore during install/uninstall operation. just need to consider one case, is it possible that user create a new .net sdk project when another project is installing package?

Reply:
Nice catch, I've updated the code to just skip it for current async operation so that it still continue to run auto restore for those projects which are not impacted by the current operation or created newly.